src/engine/websites/decorators/Common.ts

```diff
@@ -83,7 +83,7 @@ export async function FetchMangaCSS(this: MangaScraper, provider: MangaPlugin, u
     const uri = new URL(url);
     const request = new FetchRequest(uri.href);
     const data = (await FetchCSS<HTMLElement>(request, query)).shift();
-    return new Manga(this, provider, uri.pathname, extract.call(this, data));
+    return new Manga(this, provider, uri.pathname + uri.search, extract.call(this, data));
```

Comment:
The usage of `pathname` was by intention as `search` or `hash` components of an URL makes no sense to be used for resource identification purposes.
I guess some websites don't care about the correct usage of URL components.

Should `uri.hash` also be added?
Don't forget to update [document tag](https://github.com/manga-download/haruneko/blob/1c86efa10a212f20dd8d7406b836f62d60a19876/src/engine/websites/decorators/Common.ts#L75) which explicitly states that only the `pathname` is used for identification.

Reply:
I dont think its necessary to add uri.hash. I mean, uri.hash is barely used for resource identification i've only seen in Mega.nz url to separate the key from the url. But since  i dont see any bad consequences doing so we may as well add it.

As for the search for identifying a resource, thats just the old php way and we should keep it for legacy websites.

Ill modify the informations accordingly

Reply:
I made uri.search & uri.hash inclusion conditional with 2 new optional parameters. This should not affect websites at all.